### PR TITLE
Use vendor patient templates

### PR DESCRIPTION
**Pull request by emc4131**
Updated to allow for a combination of Bundle/Vendor selection when using CVU+. When the user creates a product, product tests will be created using both or either vendor & bundle patients (as chosen by the user).

Pull requests into Cypress require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable ("N/A") and :white_check_mark:.

**Submitter:**
- [x] This pull request describes why these changes were made.
- [x] Internal ticket for this PR: https://jira.mitre.org/browse/CYPRESS-448
- [x] Internal ticket links to this PR
- [x] Code diff has been done and been reviewed
- [x] Tests are included and test edge cases
- [x] Tests have been run locally and pass

**Reviewer 1:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

**Reviewer 2:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

### CODE REVIEW

File: app/models/product_test.rb [hunk 1]
```diff
@@ -202,31 +203,64 @@ def self.most_recent
...
   private
 
+  def gather_patient_ids
+    aggregate_id_list = []
+    # if it's a cvu_plus product
+    if product.cvuplus
+      # then check if vendor and/or patients are included in product
+      if product.vendor_patients
+        # If so, add appropriate vendor patient ids
+        aggregate_id_list.concat product.vendor.patients.pluck(:id)
```
Comment (lmd59) [inline app/models/product_test.rb]: This should be adding vendor patients that are associated with the selected bundle- not all vendor patients. Will cause issues with a vendor that has patients across multiple bundles. Could re-add "def vendor_patient_ids" convenience method or similar code:
product.vendor.patients.where(bundleId: bundle.id).pluck(:id)
